lib/semantics/check-omp-structure.cc

```diff
@@ -14,9 +14,82 @@
 
 #include "check-omp-structure.h"
 #include "../parser/parse-tree.h"
+#include <variant>
 
 namespace Fortran::semantics {
 
+bool OmpStructureChecker::HasInvalidCloselyNestedRegion(
+    const parser::CharBlock &source, const OmpDirectiveEnumType &set) {
+  if (ompContext_.size() && set.test(GetOmpContext().currentDirectiveEnum)) {
+    context_.Say(source,
+        "A worksharing region may not be closely nested inside a "
+        "worksharing, explicit task, taskloop, critical, ordered, atomic, or "
+        "master region."_err_en_US);
```

Comment:
Can you make the message explicitly say which of the alternatives it is nested inside? That would also make the message shorter.

Reply:
okay!

Reply:
This request will be implemented along with the check for the location of DO loops of nested regions.